ArticleTemplates/assets/js/modules/cards.js

```diff
@@ -1,96 +1,39 @@
+            initialised = true;
+            setupGlobals();
         }
     }
 
-    function articleCardsFailed() {
-        var relatedContent = document.getElementsByClassName('related-content')[0];
-
+    function getRelatedContentPosition() {
+        var relatedContent = document.querySelector('.related-content');
         if (relatedContent) {
-            relatedContent.classList.add('related-content--has-failed');       
-        }
-    }
-
-    function snapToGrid() {
-        var relatedContentList = document.getElementsByClassName('related-content__list')[0];
-
-        if (relatedContentList) {
-            setUpFlipSnap(relatedContentList);
-            window.addEventListener('resize', util.debounce(onResize.bind(null, relatedContentList), 100));
-        }
-    }
-
-    function onResize(relatedContentList) {
-        if (flipSnap && flipSnap.destroy && relatedContentList) {
-            flipSnap.destroy();
-            relatedContentList.removeAttribute('style');
-            setUpFlipSnap(relatedContentList);
+            return util.getElementOffset(relatedContent);
         }
     }
 
-    function setUpFlipSnap(relatedContentList) {
-        var container = relatedContentList.parentNode,
-            containerStyle = container.currentStyle || window.getComputedStyle(container),
-            containerWidth = container.offsetWidth - parseInt(containerStyle.paddingRight.replace('px','')) - parseInt(containerStyle.paddingLeft.replace('px',''));
-
-        // add a class with the number of child items, so we can set the widths based on that 
-        relatedContentList.classList.add('related-content__list--items-' + relatedContentList.childElementCount);
-
-        if (relatedContentList.scrollWidth > containerWidth) {
-            flipSnap = flipSnap(relatedContentList);
-
-            // Android needs to be notified of touch start / touch end so article navigation can be disabled / enabled
-            if (GU.opts.platform === 'android') {
-                relatedContentList.addEventListener('touchstart', onTouchStart);
-                relatedContentList.addEventListener('touchend', onTouchEnd);
-            }
-        }            
-    }
-
-    function onTouchStart() {
-        window.GuardianJSInterface.registerRelatedCardsTouch(true);
+    function setRelatedItemsHeight(height) {
+        document.querySelector('.related-content').style.height = height + 'px';
     }
 
-    function onTouchEnd() {
-        window.GuardianJSInterface.registerRelatedCardsTouch(false);   
-    }
-
-    function ready() {
-        if (!initialised) {
-            initialised = true;
-            setupGlobals();
-        }
+    function setupGlobals() {
+        window.getRelatedContentPosition = getRelatedContentPosition;
+        window.applyNativeFunctionCall("getRelatedContentPosition");
+        window.setRelatedItemsHeight = setRelatedItemsHeight;
+        window.applyNativeFunctionCall("setRelatedItemsHeight");
```

Comment:
If you do rename `setRelatedItemsHeight` to `setRelatedContentHeight` it'll need updating here as well.